tests/components/telegram_bot/conftest.py

```diff
@@ -83,6 +83,13 @@ def mock_external_calls():
         yield
 
 
+@pytest.fixture
+def mock_stop_polling():
+    """Mock stop_polling as it makes an external call."""
+    with patch("homeassistant.components.telegram_bot.polling.PollBot.stop_polling"):
```

Comment:
I don't believe we need to patch `stop_polling` since we already patch the application. All the application methods should already be mocks.

Reply:
It's required for the `polling_platform` fixture used by `test_polling_platform_init` because the `start_polling` is never actually called.

It's on the patched application test because there's an error raised from using the mocked class.

```
  File "/workspaces/ha_core/homeassistant/components/telegram_bot/polling.py", line 61, in stop_polling
    await self.application.updater.stop()
TypeError: object MagicMock can't be used in 'await' expression
```

Reply:
Then I think we should mock the application update stop method with an `AsyncMock`.

Reply:
it is all three methods in `stop_polling` that require `AsyncMock`